xrpl/models/transactions/batch.py

```diff
@@ -0,0 +1,149 @@
+    """Represents a Batch signer."""
+
+    account: str = REQUIRED  # type: ignore
+
+    signing_pub_key: Optional[str] = None
+
+    txn_signature: Optional[str] = None
+
+    signers: Optional[List[Signer]] = None
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True, **KW_ONLY_DATACLASS)
+class Batch(Transaction):
+    """Represents a Batch transaction."""
+
+    raw_transactions: List[Transaction] = REQUIRED  # type: ignore
+    batch_signers: Optional[List[BatchSigner]] = None
+
+    transaction_type: TransactionType = field(
+        default=TransactionType.BATCH,
+        init=False,
+    )
+
+    def __post_init__(self: Self) -> None:
+        """Called by dataclasses immediately after __init__."""
+        new_raw_transactions = []
+        for tx in self.raw_transactions:
+            if tx.has_flag(TransactionFlag.TF_INNER_BATCH_TXN):
+                new_raw_transactions.append(tx)
+            else:
+                tx_dict = tx.to_dict()
+                if "flags" not in tx_dict:
+                    tx_dict["flags"] = TransactionFlag.TF_INNER_BATCH_TXN
+                elif isinstance(tx_dict["flags"], int):
+                    tx_dict["flags"] += TransactionFlag.TF_INNER_BATCH_TXN
+                elif isinstance(tx_dict["flags"], dict):
+                    tx_dict["flags"]["TF_INNER_BATCH_TXN"] = True
+                elif isinstance(self.flags, list):
+                    tx_dict["flags"].append(TransactionFlag.TF_INNER_BATCH_TXN)
+                else:  # None
+                    tx_dict["flags"] = TransactionFlag.TF_INNER_BATCH_TXN
+                new_raw_transactions.append(Transaction.from_dict(tx_dict))
+        # This is done before dataclass locks in the frozen fields
+        # This way of editing is a bit hacky, but it's the recommended method for
+        # updating frozen fields in dataclasses
+        # https://docs.python.org/3/library/dataclasses.html#frozen-instances
+        object.__setattr__(self, "raw_transactions", new_raw_transactions)
+        super().__post_init__()
+
+    def _get_errors(self: Self) -> Dict[str, str]:
+        errors = super()._get_errors()
+        for i in range(len(self.raw_transactions)):
+            tx = self.raw_transactions[i]
+            if not tx.has_flag(TransactionFlag.TF_INNER_BATCH_TXN):
+                errors[f"raw_transactions[{i}]"] = (
+                    "RawTransaction must have tfInnerBatchTxn flag set."
+                )
+        return errors
+
+    @classmethod
+    def from_dict(cls: Type[Self], value: Dict[str, Any]) -> Self:
+        """
+        Construct a new Batch from a dictionary of parameters.
+
+        Args:
+            value: The value to construct the Batch from.
+
+        Returns:
+            A new Batch object, constructed using the given parameters.
+
+        Raises:
+            XRPLModelException: If the dictionary provided is invalid.
+        """
+        new_value = {**value}
+        new_value["raw_transactions"] = [
```

Comment:
The `__post_init__` dunder of the Batch class already initializes a `self.raw_transactions` field. Why do we need to do this manually inside the `from_dict` method?

Reply:
The `from_dict` doesn't add a `raw_transactions` field, it updates each field in `self.raw_transactions` to be within a `RawTransaction` inner object.